### PR TITLE
Expose Rendering for EntityParts

### PR DESCRIPTION
**Pull request by Platymemo**
As per:
image

Changes the responsibility of rendering of entity parts to themselves, so they can modify it as necessary.

### CODE REVIEW

**Review by Arathain — APPROVED** [review]
:+1:

**Review by OroArmor — APPROVED** [review]

**Review by maximumpower55 — APPROVED** [review]
LGTM lets get this merged.

File: library/entity/multipart/src/main/java/org/quiltmc/qsl/entity/multipart/api/EntityPart.java [hunk 1]
```diff
@@ -30,4 +38,29 @@
...
+	 * @param matrices the {@link MatrixStack matrix stack} used for rendering
+	 * @param vertices the {@link VertexConsumer vertex consumer} used for rendering
+	 * @param ownerX the {@link #getOwner() owner's} rendered X coordinate
+	 * @param ownerY the {@link #getOwner() owner's} rendered Y coordinate
+	 * @param ownerZ the {@link #getOwner() owner's} rendered Z coordinate
+	 * @param owner the {@link #getOwner() owner}
+	 * @param tickDelta progress for linearly interpolating between the previous and current game state
+	 */
+	@Environment(EnvType.CLIENT)
+	default void renderHitbox(MatrixStack matrices, VertexConsumer vertices, double ownerX, double ownerY, double ownerZ, Entity owner, float tickDelta) {
```
Comment (LambdAurora) [inline library/entity/multipart/src/main/java/org/quiltmc/qsl/entity/multipart/api/EntityPart.java:55]: Oh damn, a client-only method in a normal class?

We would scare some people with that :P
I'm not sure what other option we have though. Definitely means we won't be able to split QSL into common/client source sets, not that we were planning to.
Will take a bit to think about it, but don't take my comment as blocking.

**Review by EnnuiL — APPROVED** [review]

**Review by williambl — APPROVED** [review]
Seems like the best solution available.

File: library/entity/multipart/src/main/java/org/quiltmc/qsl/entity/multipart/api/EntityPart.java (hunk 1 above)
Comment (Platymemo) [reply library/entity/multipart/src/main/java/org/quiltmc/qsl/entity/multipart/api/EntityPart.java:55]: I think this is the best solution. We could create some kind of optional EntityPartRenderingRegistry, but I think this method is best for discoverability and ease of use.

**Comment by Arathain** [comment]
any progress on this / anything blocking this entering FCP?

**Review by LambdAurora — APPROVED** [review]

**Pull request merged**